### PR TITLE
Implement JitDefaultFill

### PR DESCRIPTION
**Pull request by briansull**
Changes and bugs fixes when changing the JitDefaultFill value from 0xff to 0xdd

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
How does the randomization of the fill value under stress happen? Is it set externally?

**Comment by briansull** [comment]
I will look into that.
I resurrected this changeset from some work I had done 6 months ago.
The randomization under Jit Stress might still need to be added.
It did uncover a couple of existing bugs that I added fixes for.

File: src/jit/compiler.cpp [hunk 1]
```diff
@@ -2194,6 +2194,16 @@ void Compiler::compDoComponentUnitTestsOnce()
         BitSetSupport::TestSuite(getAllocatorDebugOnly());
     }
 }
+
+unsigned char Compiler::compGetJitDefaultFill()
+{
+    unsigned char defaultFill = (unsigned char)JitConfig.JitDefaultFill();
+
+    // compStressCompile(STRESS_GENERIC_VARN, 50);
+
```
Comment (briansull) [inline src/jit/compiler.cpp]: To add random defaultFill values with JitStress I will add:

```

    if (compStressCompile(STRESS_GENERIC_VARN, 50))
    {
        defaultFill = info.compMethodHash();
        defaultFill ^= (defaultFill >> 12);
        defaultFill &= 0xff;
    }
```

Comment (briansull) [reply src/jit/compiler.cpp]: That will make it such that 50% time we pick a semi-random JitFill value based upon the Method Hash.

Comment (briansull) [reply src/jit/compiler.cpp]: Well we havethe lame assert around the JIT thatmake thsi difficult:

See morph.cpp line 2624:

```
void fgArgInfo::RecordStkLevel(unsigned stkLvl)
{
    assert(!IsUninitialized(stkLvl));
    this->stkLevel = stkLvl;
}

```

**Comment by briansull** [comment]
@dotnet-bot test Windows_NT x64 Checked jitstress1
@dotnet-bot test Windows_NT x64 Checked jitstress2
@dotnet-bot test Ubuntu x64 Checked jitstress1
@dotnet-bot test Ubuntu x64 Checked jitstress2
@dotnet-bot test Windows_NT x86 Checked jitstress1
@dotnet-bot test Windows_NT x86 Checked jitstress2

**Comment by briansull** [comment]
@dotnet-bot test Windows_NT x64 Checked jitstress1
@dotnet-bot test Windows_NT x64 Checked jitstress2
@dotnet-bot test Ubuntu x64 Checked jitstress1
@dotnet-bot test Ubuntu x64 Checked jitstress2
@dotnet-bot test Windows_NT x86 Checked jitstress1
@dotnet-bot test Windows_NT x86 Checked jitstress2

**Comment by briansull** [comment]
No AsmDiffs withthis change
JitStress1 and JitStress2 are both passing with Windows x64  and Ubuntu x64

@AndyAyersMS PTAL
@dotnet/jit-contrib  PTAL

File: src/jit/compiler.cpp [hunk 2]
```diff
@@ -2194,6 +2194,31 @@ void Compiler::compDoComponentUnitTestsOnce()
         BitSetSupport::TestSuite(getAllocatorDebugOnly());
     }
 }
+
+unsigned char Compiler::compGetJitDefaultFill()
```
Comment (AndyAyersMS) [inline src/jit/compiler.cpp]: Header comment block?

File: src/jit/jit.h [hunk 3]
```diff
@@ -947,9 +930,46 @@ class JitTls
...
+// Include the definition of Compiler for use by template functions
+//
 #include "compiler.h"
 
+/*****************************************************************************
+*
+*  Returns a word filled with the JITs allocator CHK fill value.
+*
+*/
+template <typename T>
```
Comment (AndyAyersMS) [inline src/jit/jit.h]: Should these new methods should use the new header comment style?

**Comment by briansull** [comment]
@AndyAyersMS - Updated comment headers from your feedback

File: src/jit/compiler.cpp [hunk 4]
```diff
@@ -2195,6 +2195,19 @@ void Compiler::compDoComponentUnitTestsOnce()
...
+// compGetJitDefaultFill:
+//
+// Return Value:
+//    An unsigned char value used to initizalize memory allocated by the JIT.
+//    The default value is taken from COMPLUS_JitDefaultFill,  if is not set
+//    the value will be 0xdd.  When JitStress is active a random value based
+//    on the method hash is used.
+//
+// Notes:
+//    Note that we can't use small values like zero, becausecwe have some
```
Comment (AndyAyersMS) [inline src/jit/compiler.cpp]: becausecwe --> because we

Comment (briansull) [reply src/jit/compiler.cpp]: fixed

**Pull request merged**